### ISSUE TITLE
python-tk@3.9: It's no longer possible to 'import tkinter' in a Python venv / virtualenv (but it works outside of virtualenv)

### ISSUE THREAD

**Issue by vlasovskikh** [Homebrew/homebrew-core#73987]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
$ brew config
HOMEBREW_VERSION: 3.0.9
ORIGIN: https://github.com/Homebrew/brew
HEAD: ed87211daa83982a6136e14d181a1550b46a0f17
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 95b3f0a3d374ce5168c7775ed2b0c980d7441901
Core tap last commit: 42 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_BROWSER: chromium
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.aW29kt0Cjd/org.macosforge.xquartz:0
HOMEBREW_EDITOR: vim
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 12.0 build 1200
Git: 2.31.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2-x86_64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
XQuartz: 2.7.11 => /opt/X11
```

```
$ brew doctor
Your system is ready to brew.
```

### 

- [X] The `brew doctor` above contains no "Warning" lines.

### What were you trying to do (and why)?

As a Python Tk GUI app developer I want to use the [tkinter](https://docs.python.org/3/library/tkinter.html) module from the Python 3.9 standard library. I follow a common practice and develop my Python code in a Python virtual environment. I was trying to run my Python code that has `import tkinter` lines within a Python virtual environment created using a Homebrew-installed Python 3.9. It used to work fine before the introduction of `python-tk` package in Homebrew on 2021-03-26.

### What happened (include all command output)?

I get an `ImportError` for `_tkinter` when running a file `a.py` that contains `import tkinter` within a Python virtual environment based on a Homebrew-installed Python 3.9 (see the steps to reproduce):

```
(venv) $ python a.py
Traceback (most recent call last):
  File "~/tkinter-test/a.py", line 1, in <module>
    import tkinter
  File "/usr/local/Cellar/python@3.9/3.9.2_3/Frameworks/Python.framework/Versions/3.9/lib/python3.9/tkinter/__init__.py", line 37, in <module>
    import _tkinter # If this fails your Python may not be configured for Tk
ModuleNotFoundError: No module named '_tkinter'
```

At the same time, this file runs successfully using the main Python 3.9 interpreter (not from the virtual environment):

```
(venv) $ /usr/local/opt/python@3.9/bin/python3.9 a.py
```

I get no error message and exit code 0.

### What did you expect to happen?

I expected my Python file containing `import tkinter` to finish successfully with no output and exit code 0:

```
(venv) $ python a.py
```

### Step-by-step reproduction instructions (by running `brew` commands)

Steps to reproduce:

1. Install `python-tk`

```
$ brew install python-tk
==> Downloading https://homebrew.bintray.com/bottles/python-tk%403.9-3.9.2.big_sur.bottle.tar.gz
Already downloaded: ~/Library/Caches/Homebrew/downloads/eab8fe791f6b68eebffc5db526104d62fb423aa54049ed668d3de450da453827--python-tk@3.9-3.9.2.big_sur.bottle.tar.gz
==> Pouring python-tk@3.9-3.9.2.big_sur.bottle.tar.gz
🍺  /usr/local/Cellar/python-tk@3.9/3.9.2: 5 files, 132.4KB
```

2. Create a test directory `tkinter-test` and create a Python virtual environment in `tkinter-test/venv` using Homebrew-installed Python 3.9

```
$ mkdir ~/tkinter-test
$ cd ~/tkinter-test/
$ /usr/local/opt/python@3.9/bin/python3.9 -m venv venv
```

3. Activate the created Python virtual environment

```
$ source venv/bin/activate
```

4. Create a file `a.py` containing one line of Python code: `import tkinter`

```
(venv) $ vim a.py
(venv) $ cat a.py
import tkinter
```

5. Run the created Python file using `python` from the created virtual environment

```
(venv) $ python a.py
```

**Comment by carlocab** [comment]
CC @Bo98

**Comment by Bo98** [comment]
Not entirely sure what the standard way of handling this is - I'll have a look.

**Comment by vlasovskikh** [comment]
The problem is with the fact that the new formula `python-tk@3.9` installs `_tkinter` binary Python module into the `site-packages` directory dedicated to non-standard libraries, while both `tkinter` and `_tkinter` are in fact a part of the Python 3.9 standard library:

```
$ /usr/local/opt/python@3.9/bin/python3.9
Python 3.9.2 (default, Mar 25 2021, 03:35:43)
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import _tkinter
>>> _tkinter
<module '_tkinter' from '/usr/local/lib/python3.9/site-packages/_tkinter.cpython-39-darwin.so'>
>>> '/usr/local/lib/python3.9/site-packages' in sys.path
True
```

In a a Python virtual environment based on Python 3.9:

```
(venv) $ python
Python 3.9.2 (default, Mar 25 2021, 03:35:43)
[Clang 12.0.0 (clang-1200.0.32.29)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys
>>> '/usr/local/lib/python3.9/site-packages' in sys.path
False
>>> import tkinter
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Cellar/python@3.9/3.9.2_3/Frameworks/Python.framework/Versions/3.9/lib/python3.9/tkinter/__init__.py", line 37, in <module>
    import _tkinter # If this fails your Python may not be configured for Tk
ModuleNotFoundError: No module named '_tkinter'
```

**Comment by Bo98** [comment]
I could potentially plant a `pth` file in `python@3.9`'s prefix that points to the `python-tk@3.9` formula. A bit of a hack but might be best option.

**Comment by Bo98** [comment]
My proposal is:

* Move `sitecustomize.py` into `python@3.9`'s prefix from general site-packages (bonus: this fixes `sys.base_prefix` within venvs).
* Update `python-tk@3.9` to install the file outside of the regular search path (e.g. `libexec`).
* Adjust `sitecustomize.py` to search for `python-tk@3.9` and to add that prefix to the search path.

I think that makes sense? Having a `python-tk@3.9` reference within `python@3.9` is a little annoying - but I suppose there's a bit of that anyway to disable the `_tkinter` build.

**Comment by Bo98** [comment]
Thanks for the report @vlasovskikh. I've merged #73994 which should hopefully improve things here.

**Comment by vlasovskikh** [comment]
@Bo98 Thank you for your quick fix! I've tested it with my application and I've checked how the updated `sys.path` works for virtualenvs and the main interpreter. Everything works well for me.
